unittests/test_modules.py

```diff
@@ -401,3 +403,55 @@ def test_mapping_deep_cycle(self):
         }
         self.assertRaisesRegex(EnvironError, 'm0->m1->m3->m4->m2->m1',
                                self.modules_system.load_module, 'm0')
+
+    def test_mapping_from_file_simple(self):
+        self.fp.write('m1:m2  m3   m3\n'
+                      'm2 : m4 \n'
+                      ' #m5: m6\n'
+                      '\n'
+                      'm5: m6 # Inline comment')
+        reference_map = {
+            'm1': ['m2', 'm3'],
+            'm2': ['m4'],
+            'm5': ['m6']
+        }
+        self.fp.close()
+        self.modules_system.load_mapping_from_file(self.fp.name)
+        self.assertEqual(reference_map, self.modules_system.module_map)
+
+    def test_mapping_from_file_repeated_key(self):
```

Comment:
I think this test can be merged with the above. Simply, add a duplicate key in the above dictionary and update your `reference_map`.